File: app/src/main/java/com/example/sharewhatyoucanproject/HomescreenActivity.java
package com.example.sharewhatyoucanproject;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.view.View;

public class HomescreenActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_homescreen);
        //donor action
        Button button = (Button) findViewById(R.id.button_first);
        button.setOnClickListener(new View.OnClickListener()
        {
            public void onClick(View v) {
                // Do something in response to button click
                startActivity(new Intent(HomescreenActivity.this, DonorActivity.class));
            }
        });

        //receiver action
        Button button2 = (Button) findViewById(R.id.button_second);
        button2.setOnClickListener(new View.OnClickListener()
        {
            public void onClick(View v) {
                // Do something in response to button click
                startActivity(new Intent(HomescreenActivity.this, DashboardActivity.class));
            }
        });



    }
}

File: app/src/main/java/com/example/sharewhatyoucanproject/HomescreenActivity.kt
package com.example.sharewhatyoucanproject

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.sharewhatyoucanproject.R
import android.content.Intent
import com.example.sharewhatyoucanproject.HomescreenActivity
import com.example.sharewhatyoucanproject.MainActivity
import android.widget.EditText
import android.app.ProgressDialog
import android.view.View
import android.widget.Button
import com.example.sharewhatyoucanproject.DonorActivity
import com.example.sharewhatyoucanproject.DashboardActivity

class HomescreenActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_homescreen)
        //donor action
        val button = findViewById<View>(R.id.button_first) as Button
        button.setOnClickListener { // Do something in response to button click
            startActivity(Intent(this@HomescreenActivity, DonorActivity::class.java))
        }

        //receiver action
        val button2 = findViewById<View>(R.id.button_second) as Button
        button2.setOnClickListener { // Do something in response to button click
            startActivity(Intent(this@HomescreenActivity, DashboardActivity::class.java))
        }
    }
}